packages/blocks/src/__internal__/utils/block-range.ts

```diff
@@ -41,13 +50,19 @@ export type BlockRange = {
 };
 
 type ExtendBlockRange = {
-  type: 'Title';
+  type: 'Title' | 'DatabaseTitle' | 'DatabaseRichText';
   startOffset: number;
   endOffset: number;
+  /**
+   * database rich-text
+   */
+  databaseCellPos?: DatabaseCellPos;
   /**
    * Only one model, the page model
+   *
+   * Two models for database rich-text
    */
-  models: [BaseBlockModel];
+  models: [BaseBlockModel] | [BaseBlockModel, BaseBlockModel];
 };
```

Comment:
You should use the unions type to ensure type-safe, for example

```ts
type ExtendBlockRange = {
  type: 'Title';
  startOffset: number;
  endOffset: number;
  models: [BaseBlockModel];
} | 
{
  type: 'Database';
  cellPos: number;
  startOffset: number;
  endOffset: number;
  models: [BaseBlockModel, BaseBlockModel];
}

```

And I think it can merge `DatabaseTitle` and `DatabaseRichText` into a `Databse` type.